Hazelcast.Net/Hazelcast.Client/Statistics.cs

```diff
@@ -0,0 +1,293 @@
+        private const string EmptyStatValue = "";
+
+        private readonly HazelcastClient _client;
+        private readonly bool _enabled;
+        private readonly int _period;
+        private CancellationTokenSource _heartbeatToken;
+
+        private readonly AtomicBoolean _live = new AtomicBoolean(false);
+
+        private readonly IDictionary<string, Func<string>> _gauges = new Dictionary<string, Func<string>>();
+
+        private volatile Address _ownerAddress;
+
+        public Statistics(HazelcastClient client)
+        {
+            _client = client;
+            _enabled = EnvironmentUtil.ReadBool("hazelcast.client.statistics.enabled") ?? false;
+            _period = EnvironmentUtil.ReadInt("hazelcast.client.statistics.period.seconds") ?? 3;
+        }
+
+        public void Start()
+        {
+            if (!_enabled || !_live.CompareAndSet(false, true))
+            {
+                return;
+            }
+
+            RegisterMetrics();
+
+            _heartbeatToken = new CancellationTokenSource();
+
+            _client.GetClientExecutionService().ScheduleWithFixedDelay(PeriodicStatisticsSendTask, _period, _period,
+                TimeUnit.Seconds, _heartbeatToken.Token);
+        }
+
+        public void Destroy()
+        {
+            if (!_live.CompareAndSet(true, false))
+            {
+                return;
+            }
+
+            _heartbeatToken.Cancel();
+        }
+
+        private ClientConnection GetOwnerConnection()
+        {
+            var ownerAddress = _client.GetClientClusterService().GetOwnerConnectionAddress();
+            var clientConnection = _client.GetConnectionManager().GetConnection(ownerAddress);
+            if (clientConnection == null)
+            {
+                return null;
+            }
+
+            if (clientConnection.ConnectedServerVersionInt >= SinceVersion)
+            {
+                return clientConnection;
+            }
+
+            // do not print too many logs if connected to an old version server
+            if (_ownerAddress == null || !_ownerAddress.Equals(ownerAddress))
+            {
+                if (Logger.IsFinestEnabled())
+                {
+                    Logger.Finest(string.Format(
+                        "Client statistics can not be sent to server {0} since, " +
+                        "connected owner server version is less than the minimum supported server version {1}",
+                        ownerAddress, SinceVersionString));
+                }
+            }
+
+            // cache the last connected server address for decreasing the log prints
+            _ownerAddress = ownerAddress;
+            return null;
+        }
+
+        private void PeriodicStatisticsSendTask()
+        {
+            var ownerConnection = GetOwnerConnection();
+            if (null == ownerConnection)
+            {
+                Logger.Finest("Can not send client statistics to the server. No owner connection.");
```

Comment:
`GetOwnerConnection` method does not print repetitive logs if the owner does not support statistics. However, this method prints repetitive logs when server does not support statisticks, right? It kind of reverses what `GetOwnerConnection` does about logging.

Also, do you need `Logger.isFinestEnabled` check here like in `GetOwnerConnection`?